libglusterfs/src/glusterfs/iobuf.h

```diff
@@ -140,7 +142,8 @@ iobuf_to_iovec(struct iobuf *iob, struct iovec *iov);
 
 #define iobuf_ptr(iob) ((iob)->ptr)
 #define iobpool_default_pagesize(iobpool) ((iobpool)->default_page_size)
-#define iobuf_pagesize(iob) (iob->iobuf_arena->page_size)
+#define iobuf_pagesize(iob)                                                    \
+    (iob->iobuf_arena ? iob->iobuf_arena->page_size : iob->page_size)
```

Comment:
We could always initialize `iob->page_size` and use it here, independently if it comes from an arena or not.

Reply:
It is difficult to initialize always because with the help of iob->page_size I am taking decision the memory block was allocated by standard allocation not MALLO/CALLOC.

Reply:
we can use exactly the same but compare with `USE_IOBUF_POOL_IF_SIZE_GREATER_THAN` instead of 0.

Reply:
or we could use `iobuf_arena` field. It should only be set when the iobuf comes from an arena.

Reply:
done